src/main/java/uk/gov/hmcts/ccd/domain/service/getevents/AuthorisedGetEventsOperation.java

```diff
@@ -27,31 +25,31 @@
     private final GetEventsOperation getEventsOperation;
     private final AccessControlService accessControlService;
     private final CaseDefinitionRepository caseDefinitionRepository;
-    private final UserRepository userRepository;
+    private final CaseAccessService caseAccessService;
 
     public AuthorisedGetEventsOperation(@Qualifier("classified") GetEventsOperation getEventsOperation,
                                         @Qualifier(CachedCaseDefinitionRepository.QUALIFIER)
                                             CaseDefinitionRepository caseDefinitionRepository,
                                         AccessControlService accessControlService,
-                                        @Qualifier(CachedUserRepository.QUALIFIER) UserRepository userRepository) {
+                                        CaseAccessService caseAccessService) {
 
         this.getEventsOperation = getEventsOperation;
         this.accessControlService = accessControlService;
         this.caseDefinitionRepository = caseDefinitionRepository;
-        this.userRepository = userRepository;
+        this.caseAccessService = caseAccessService;
     }
 
     @Override
     public List<AuditEvent> getEvents(CaseDetails caseDetails) {
-
         final List<AuditEvent> events = getEventsOperation.getEvents(caseDetails);
-
-        return secureEvents(caseDetails.getCaseTypeId(), events);
+        return secureEvents(caseDetails.getCaseTypeId(), caseDetails.getId(), events);
     }
 
     @Override
     public List<AuditEvent> getEvents(String jurisdiction, String caseTypeId, String caseReference) {
-        return secureEvents(caseTypeId, getEventsOperation.getEvents(jurisdiction, caseTypeId, caseReference));
+        List<AuditEvent> auditEvents = getEventsOperation.getEvents(jurisdiction, caseTypeId, caseReference);
+        String caseId = auditEvents != null && auditEvents.size() > 0 ? auditEvents.get(0).getCaseDataId() : "";
```

Comment:
we could simplify this by considering that audit event's for a case cannot be null. There will be always at least on event in the case history of a case, the creation event. I'd say we can assume that's the case and remove these extra checks.

Reply:
Done